src/main.cpp

```diff
@@ -5827,8 +5827,20 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
     else if (strCommand == NetMsgType::BLOCKTXN && !fImporting && !fReindex) // Ignore blocks received while importing
     {
         BlockTransactions resp;
+        int nSize = vRecv.size();
         vRecv >> resp;
 
+        BlockMap::iterator mi = mapBlockIndex.find(resp.blockhash);
```

Comment:
Do you mean I need to add a LOCK(cs_main) before this line? I'm still trying to get my head around locks... what is the risk without obtaining a lock first please?